kornia/augmentation/_2d/mix/transplantation.py

```diff
@@ -0,0 +1,261 @@
+from typing import Any, Dict, List, Optional, Sequence, Tuple, Union
+
+import torch
+
+from kornia.augmentation._2d.mix.base import MixAugmentationBaseV2
+from kornia.augmentation.utils import _validate_input_dtype
+from kornia.constants import DataKey, DType
+from kornia.core import Tensor, tensor
+
+__all__ = ["RandomTransplantation"]
+
+
+class RandomTransplantation(MixAugmentationBaseV2):
+    r"""RandomTransplantation augmentation.
+
+    .. image:: _static/img/RandomTransplantation.png
+
+    Randomly transplant (copy and paste) image features and corresponding segmentation masks between images in a batch.
+    The transplantation transform works as follows:
+
+        1. Based on the parameter `p`, a certain number of images in the batch are selected as acceptor of a
+           transplantation.
+        2. For each acceptor, the image below in the batch is selected as donor (via circling: :math:`i - 1 \mod B`).
+        3. From the donor, a random label is selected and the corresponding image features and segmentation mask are
+           transplanted to the acceptor.
+
+    The augmentation is described in `Semantic segmentation of surgical hyperspectral images under geometric domain
+    shifts` :cite:`sellner2023semantic`.
+
+    Args:
+        excluded_labels: sequence of labels which should not be transplanted from a donor. This can be useful if only
+          parts of the image are annotated and the non-annotated regions (with a specific label index) should be
+          excluded from the augmentation. If no label is left in the donor image, nothing is transplanted.
+        p: probability for applying an augmentation to an image. This parameter controls how many images in a batch
+          receive a transplant.
+        p_batch: probability for applying an augmentation to a batch. This param controls the augmentation
+          probabilities batch-wise.
+        data_keys: the input type sequential for applying augmentations.
+          Accepts "input", "mask".
+
+    Note:
+        - This augmentation requires that segmentation masks are available for all images in the batch and that at
+          least some objects in the image are annotated.
+        - This implementation works for arbitrary spatial dimensions including 2D and 3D images.
```

Comment:
Not sure. IIRC, the base class would check if all the input are BCHW or not. Probably it is better to make it only support 2D for now. Or if you got time, we may have the 3D MixAugmentationBase.

Reply:
The augmentation itself is independent of the spatial dimension since it relies only on the segmentation mask. The current version of the implementation is also independent of the spatial dimensions since I overwrite the `forward()` function. I also changed the test to cover at least 2D, 3D and 4D input.